### DISCUSSION TITLE
OneStepCheckout Not Compatible with OpenMage 1.9

### DISCUSSION THREAD

**Discussion by djixas** [OpenMage/magento-lts#3233] [Questions & Answers]
### Steps to reproduce (*)

1. Add any items to the shopping cart at any onestecpechkout site
2. Go to checkout and try to adjust quantity, it will not work

### Expected result (*)

1. It should work

### Actual result (*)

1.  As per OneStepCheckout investigation:
2. There seems to be a issue with your prototype.js library on the site as it does not recognise the valid selector $$('div.payment-methods input[name^=payment\[method\]]') and instead gives errors . 
 
ototype.js:5265 Uncaught Error: Syntax error, unrecognized expression: div.payment-methods input[name^=payment[method]]
    at Sizzle.error (prototype.js:5265:8)
    at tokenize (prototype.js:5800:11)
    at Sizzle.select (prototype.js:6160:20)
    at Sizzle (prototype.js:4770:9)
    at Object.select (prototype.js:6296:27)
    at window.$$ (prototype.js:4478:29)
    at <anonymous>:1:1
S
 
while testing with clean prototype.js this is perfectly valid selector and results with elements found or with [] empty array if nothing is found. If I swap out the prototypejs library to unmodified version (as openmage seems to change it ) on runtime while testing your site with devtools , then it works just as expected.

OneStepCheckout support replied also: So I suggest you to use plain prototypejs latest version and validate why openmage changes that and if it affects your site and if it is safe to use unmodified prototypejs on your site. .

**Comment by fballiano** [comment]
which version of openmage 19 are you running?

"use the latest version of prototypejs" 🤣 it was literally released a milion years ago...

**Comment by pquerner** [comment]
I can change the qty of an item just fine on your page, in the checkout cart or minicart. No errors in the console.log.

**Comment by addison74** [comment]
I confirm this issue in the onepage checkout page, but we don't support 3rd party extensions.

**Comment by fballiano** [comment]
anyway, these vendors... is this the "support" they give? update prototypejs? really? my. god.

**Comment by addison74** [comment]
A golden rule says not to modify something that works. In general, those who have little knowledge of programming, immediately update to a new version without checking its functionality intensively. I can personally say that I have variants that will be moved to production, where 2-3 months are expected until the "good to publish" stamp is put.

In thiscase the cause may be found, but I doubt it is from the prototype.js library. He can put back a version from the last Magento and see what happens. If it works then he can find out the cause.

I will move the reporting to Q&A discussions because it is not confirmed in the OpenMage checkout.

**Reply by djixas** [reply]
> A golden rule says not to modify something that works. In general, those who have little knowledge of programming, immediately update to a new version without checking its functionality intensively. I can personally say that I have variants that will be moved to production, where 2-3 months are expected until the "good to publish" stamp is put.
> 
> In thiscase the cause may be found, but I doubt it is from the prototype.js library. He can put back a version from the last Magento and see what happens. If it works then he can find out the cause.
> 
> I will move the reporting to Q&A discussions because it is not confirmed in the OpenMage checkout.

Hi,

I can confirm that when using magento-mirror-1.9.4.5\js\prototype/prototype.js, it works, opened beyond compare and there are differences, I wonder how essential is 1.7.3 vs 1.7 in original magento archive?

![bc](https://github.com/OpenMage/magento-lts/assets/12409425/3f451bd0-75c0-4b62-a9ee-1a497adb94e6)
